streams/shared/src/main/scala/zio/stream/ZChannel.scala

```diff
@@ -1293,6 +1307,47 @@ sealed trait ZChannel[-Env, -InErr, -InElem, -InDone, +OutErr, +OutElem, +OutDon
       ZIO.suspendSucceed(interpret(exec.run().asInstanceOf[ChannelState[Env, OutErr]]))
     }
 
+  final def toPullInAlt(
+    scope: => Scope
+  )(implicit trace: Trace): ZIO[Env, Nothing, ZIO[Env, Either[OutErr, OutDone], OutElem]] =
+    ZIO.uninterruptible {
+      val exec = new ChannelExecutor[Env, InErr, InElem, InDone, OutErr, OutElem, OutDone](
+        () => self,
+        null,
+        ZIO.identityFn[URIO[Env, Any]]
+      )
+      ZIO.environmentWithZIO[Env] { environment =>
+        scope.addFinalizerExit { exit =>
+          val finalizer = exec.close(exit)
+          if (finalizer ne null) finalizer.provideEnvironment(environment)
```

Comment:
@regiskuckaertz @kyri-petrou @hearnadam  

Why are we testing is the `finalizer` is not `null` here? It doesn't seem like `exec.close(exit)` can return `null`
Shouldn't it be `environment ne null`? 🤔

Reply:
Yeah good question. Although I don't think it's possible for either the `finalizer` or the `environment` to be `null` 🤔

Reply:
It will be the current environment (this is always the case, the environment might contain more services than just `R`) without any requirement that it will contain any services. So at the very "minimum" it will be `ZEnvironment.empty`